<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>957</width>
    <height>443</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Upper Monitor</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>10</y>
      <width>661</width>
      <height>161</height>
     </rect>
    </property>
    <property name="title">
     <string/>
    </property>
    <widget class="QLabel" name="label_14">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>100</y>
       <width>51</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>角度：</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_Angle">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>90</y>
       <width>81</width>
       <height>33</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_17">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>100</y>
       <width>68</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>角度方向：</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_DirAngle">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>90</y>
       <width>81</width>
       <height>33</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_13">
     <property name="geometry">
      <rect>
       <x>390</x>
       <y>50</y>
       <width>71</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>面积阈值：</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_AreaMin">
     <property name="geometry">
      <rect>
       <x>460</x>
       <y>40</y>
       <width>91</width>
       <height>33</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_ThreVal">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>40</y>
       <width>81</width>
       <height>33</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>51</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>阈值：</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_BlurSize">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>40</y>
       <width>81</width>
       <height>33</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>50</y>
       <width>71</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>滤波：</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_12">
     <property name="geometry">
      <rect>
       <x>390</x>
       <y>100</y>
       <width>68</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>距离：</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_Distance">
     <property name="geometry">
      <rect>
       <x>460</x>
       <y>90</y>
       <width>91</width>
       <height>33</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_4">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>10</y>
      <width>341</width>
      <height>161</height>
     </rect>
    </property>
    <property name="title">
     <string/>
    </property>
    <widget class="QPushButton" name="pushButton_OpenCam">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>30</y>
       <width>111</width>
       <height>51</height>
      </rect>
     </property>
     <property name="text">
      <string>打开相机</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_OpenPort">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>90</y>
       <width>111</width>
       <height>51</height>
      </rect>
     </property>
     <property name="text">
      <string>打开串口</string>
     </property>
    </widget>
    <widget class="QComboBox" name="comboBox_SerialPort">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>90</y>
       <width>121</width>
       <height>51</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>COM1</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>COM2</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>COM3</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>COM4</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>COM5</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>COM6</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>COM7</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>COM8</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>ttyUSB0</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>ttyUSB1</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>ttyUSB2</string>
      </property>
     </item>
    </widget>
    <widget class="QComboBox" name="comboBox_CameraPath">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>121</width>
       <height>51</height>
      </rect>
     </property>
     <property name="currentText">
      <string>0</string>
     </property>
     <item>
      <property name="text">
       <string>0</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>1</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>2</string>
      </property>
     </item>
    </widget>
    <widget class="QPushButton" name="pushButton_TakePhoto">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>260</x>
       <y>30</y>
       <width>61</width>
       <height>51</height>
      </rect>
     </property>
     <property name="text">
      <string>拍照</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_Calibrate">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>260</x>
       <y>90</y>
       <width>61</width>
       <height>51</height>
      </rect>
     </property>
     <property name="text">
      <string>标定</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>180</y>
      <width>931</width>
      <height>211</height>
     </rect>
    </property>
    <property name="title">
     <string/>
    </property>
    <widget class="QLabel" name="label">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>480</x>
       <y>20</y>
       <width>91</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>原图</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_Frame">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>480</x>
       <y>50</y>
       <width>201</width>
       <height>151</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>101</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>矫正图</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>710</x>
       <y>20</y>
       <width>101</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>原图（灰度图）</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>250</x>
       <y>20</y>
       <width>121</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>原图（二值化）</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_Frame_resized">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>50</y>
       <width>201</width>
       <height>151</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="label_Frame_gray">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>710</x>
       <y>50</y>
       <width>201</width>
       <height>151</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="label_Frame_binary">
     <property name="geometry">
      <rect>
       <x>250</x>
       <y>50</y>
       <width>201</width>
       <height>151</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
   </widget>
   <zorder>groupBox_3</zorder>
   <zorder>groupBox_4</zorder>
   <zorder>groupBox</zorder>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>957</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>pushButton_OpenCam</tabstop>
  <tabstop>pushButton_TakePhoto</tabstop>
  <tabstop>pushButton_Calibrate</tabstop>
  <tabstop>pushButton_OpenPort</tabstop>
  <tabstop>lineEdit_ThreVal</tabstop>
  <tabstop>lineEdit_BlurSize</tabstop>
  <tabstop>lineEdit_AreaMin</tabstop>
  <tabstop>lineEdit_Angle</tabstop>
  <tabstop>lineEdit_Distance</tabstop>
  <tabstop>lineEdit_DirAngle</tabstop>
  <tabstop>comboBox_SerialPort</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
